### PR TITLE
Reuse the same repository so that ITs can access maven snapshot jars from the repository

### PR DESCRIPTION
**Pull request by gnodet**
Supersedes #1276

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ def buildOs = 'linux'
 def buildJdk = '11'
 def buildMvn = '3.8.x'
 def runITsOses = ['linux']
-def runITsJdks = ['11', '17', '21']
+def runITsJdks = ['jdk_11_latest', 'jdk_17_latest', 'jdk_21_latest']
```
Comment (slawekjaranowski) [inline Jenkinsfile:26]: We should update in https://github.com/apache/maven-jenkins-env/blob/master/vars/jenkinsEnv.groovy `jdkFromVersion`

Comment (gnodet) [reply Jenkinsfile:26]: Ah, nice, I actually had no idea where those values were coming from :-)

**Comment by gnodet** [comment]
@slawekjaranowski it seems to be working now :-)

**Comment by gnodet** [comment]
Raised https://github.com/apache/maven-integration-testing/pull/314

Note that the IT will probably break after starting the Maven release process, as the version will change.  After the released is published, the IT should use the released version of Maven and the whole problem will be over.

**Comment by slawekjaranowski** [comment]
source branch deleted

**Pull request merged**